Extension/src/LanguageServer/configurations.ts

```diff
@@ -751,6 +751,27 @@ export class CppProperties {
 
             // TODO: Add other squiggles.
 
+            // Check if intelliSenseMode and compilerPath are compatible
+            if (os.platform() === 'win32') {
+                // cl.exe is only available on Windows
+                const intelliSenseModeStart: number = curText.search(/\s*\"intelliSenseMode\"\s*:\s*\"/);
+                if (intelliSenseModeStart !== -1) {
+                    const intelliSenseModeValueStart: number = curText.indexOf('"', curText.indexOf(":", intelliSenseModeStart));
+                    const intelliSenseModeValueEnd: number = intelliSenseModeStart === -1 ? -1 : curText.indexOf('"', intelliSenseModeValueStart + 1) + 1;
+
+                    // cl.exe and msvc mode should be used together
+                    let compilerPathAndArgs: util.CompilerPathAndArgs = util.extractCompilerPathAndArgs(this.CurrentConfiguration.compilerPath);
```

Comment:
Can this check be factored out into a function that does something like:

`isCompilerIntelliSenseModeCompatible(osPlatform: string): boolean { ... }`

We would need to reuse the code that validates inputs for both UI and JSON later on.

Reply:
Added